ledger/db/src/mint_tx_store.rs

```diff
@@ -0,0 +1,628 @@
+pub const BLOCK_INDEX_BY_MINT_TX_NONCE_DB_NAME: &str = "mint_tx_store:block_index_by_mint_tx_nonce";
+
+/// A list of mint-txs that can be prost-encoded. This is needed since that's
+/// the only way to encode a Vec<MintTx>.
+#[derive(Clone, Message)]
+pub struct MintTxList {
+    #[prost(message, repeated, tag = "1")]
+    pub mint_txs: Vec<MintTx>,
+}
+
+#[derive(Clone)]
+pub struct MintTxStore {
+    /// MintTxs by block.
+    mint_txs_by_block: Database,
+
+    /// block index by MintTx nonce.
+    block_index_by_mint_tx_nonce: Database,
+}
+
+impl MintTxStore {
+    /// Opens an existing MintTxStore.
+    pub fn new(env: &Environment) -> Result<Self, Error> {
+        Ok(MintTxStore {
+            mint_txs_by_block: env.open_db(Some(MINT_TXS_BY_BLOCK_DB_NAME))?,
+            block_index_by_mint_tx_nonce: env
+                .open_db(Some(BLOCK_INDEX_BY_MINT_TX_NONCE_DB_NAME))?,
+        })
+    }
+
+    /// Creates a fresh MintTxStore.
+    pub fn create(env: &Environment) -> Result<(), Error> {
+        env.create_db(Some(MINT_TXS_BY_BLOCK_DB_NAME), DatabaseFlags::empty())?;
+        env.create_db(
+            Some(BLOCK_INDEX_BY_MINT_TX_NONCE_DB_NAME),
+            DatabaseFlags::empty(),
+        )?;
+        Ok(())
+    }
+
+    /// Get mint txs in a given block.
+    pub fn get_mint_txs_by_block_index(
+        &self,
+        block_index: u64,
+        db_transaction: &impl Transaction,
+    ) -> Result<Vec<MintTx>, Error> {
+        let mint_txs: MintTxList =
+            decode(db_transaction.get(self.mint_txs_by_block, &u64_to_key_bytes(block_index))?)?;
+        Ok(mint_txs.mint_txs)
+    }
+
+    /// Write mint txs in a given block.
+    pub fn write_mint_txs(
+        &self,
+        block_index: u64,
+        mint_txs: &[MintTx],
+        mint_config_store: &MintConfigStore,
+        db_transaction: &mut RwTransaction,
+    ) -> Result<(), Error> {
+        let block_index_bytes = u64_to_key_bytes(block_index);
+
+        // Store the list of MintTxs.
+        let mint_tx_list = MintTxList {
+            mint_txs: mint_txs.to_vec(),
+        };
+
+        db_transaction.put(
+            self.mint_txs_by_block,
+            &block_index_bytes,
+            &encode(&mint_tx_list),
+            WriteFlags::NO_OVERWRITE, // We should not be updating existing blocks
+        )?;
+
+        // For each mint transaction, we need to locate the matching mint configuration
+        // and update the total minted count. We also need to ensure the nonce is
+        // unique.
+        for mint_tx in mint_txs {
+            // Update total minted.
+            let active_mint_config =
+                mint_config_store.get_active_mint_config_for_mint_tx(mint_tx, db_transaction)?;
+
+            let new_total_minted = active_mint_config.total_minted.checked_add(mint_tx.prefix.amount).expect("shouldn't have failed because get_active_mint_config_for_mint_tx guards against this");
```

Comment:
Optional/Equivalent: use `ok_or` rather than panicking/`expect`
```suggestion
            let new_total_minted = active_mint_config
                .total_minted
                .checked_add(mint_tx.prefix.amount)
                .ok_or(Error::NotFound)?;
```